go/vt/vtctl/workflow/materializer.go

```diff
@@ -290,6 +290,15 @@ func (mz *materializer) deploySchema() error {
 		}
 	}
 
+	// Check if any table being moved is already non-empty in the target keyspace.
+	// Skip this check for multi-tenant migrations.
+	if !mz.IsMultiTenantMigration() {
```

Comment:
A nice future addition would be to check for any existing tenant data in each table on the target too.